crates/oxc_transformer/src/jsx/refresh.rs

```diff
@@ -334,18 +326,27 @@ impl<'a, 'ctx> Traverse<'a> for ReactRefresh<'a, 'ctx> {
         arguments.insert(0, Argument::from(id_binding.create_read_expression(ctx)));
 
         let binding = BoundIdentifier::from_binding_ident(&binding_identifier);
-        self.extra_statements.entry(id_binding.symbol_id).or_insert(ctx.ast.vec()).push(
-            ctx.ast.statement_expression(
-                SPAN,
-                ctx.ast.expression_call(
-                    SPAN,
-                    binding.create_read_expression(ctx),
-                    NONE,
-                    arguments,
-                    false,
-                ),
-            ),
-        );
+        let callee = binding.create_read_expression(ctx);
+        let expr = ctx.ast.expression_call(SPAN, callee, NONE, arguments, false);
+        let statement = ctx.ast.statement_expression(SPAN, expr);
+
+        let mut target_ancestor = Ancestor::None;
+        for ancestor in ctx.ancestors() {
+            if !matches!(
+                ancestor,
+                Ancestor::ExportNamedDeclarationDeclaration(_)
+                    | Ancestor::ExportDefaultDeclarationDeclaration(_)
+            ) {
+                break;
+            }
+            target_ancestor = ancestor;
+        }
```

Comment:
Ok! 

As you mentioned above the 3rd choice, we should consider this case as well when you are implementing. Here are searching the parent of the current node if it is an `ExportNamedDeclaration` or an `ExportDefaultDeclaration`, otherwise we use `Function::address`. If we can get the current address by `Ancestor`, I think supporting such as this case will be easier.

For javascript code above this case:

```js
function x() {}
export function x() {}
export default function x() {}
```js

The logic is in `exit_function`, we need to find if the parent is `ExportNamedDeclaration` or `ExportDefaultDeclaration`, otherwise we return `Function::address` because Function is a `Statement::Function` actually.

Hope I have explained clearly!

Reply:
Thanks. I think I understand. In that case, do you need a loop? Wouldn't just checking the parent work?

```rs
match ctx.parent() {
    ancestor @ Ancestor::ExportNamedDeclarationDeclaration(_)
    | Ancestor::ExportDefaultDeclarationDeclaration(_) => {
        self.ctx.statement_injector.insert_after(&ancestor, statement);
    }
    _ => self.ctx.statement_injector.insert_after(func, statement),
}
```

Or, if we merge #6891, then:

```rs
let address = match ctx.parent() {
    Ancestor::ExportNamedDeclarationDeclaration(decl) => decl.address(),
    Ancestor::ExportDefaultDeclarationDeclaration(decl) => decl.address(),
    _ => func.address(),
};
self.ctx.statement_injector.insert_after(&address, statement);
```

Reply:
> Thanks. I think I understand. In that case, do you need a loop? Wouldn't just checking the parent work?

Oh, yes, I wrote quickly so that I forget something impossible